plugins/module_utils/foreman_helper.py

```diff
@@ -816,7 +817,12 @@ def _lookup_entity(self, identifier, entity_spec, params=None):
         try:
             if 'scope' in entity_spec:
                 for scope in entity_spec['scope']:
-                    params.update(self.scope_for(scope))
+                    params.update(self.scope_for(scope, _scope_key(resource_type, scope)))
+            if 'optional_scope' in entity_spec:
+                for optional_scope in entity_spec['optional_scope']:
```

Comment:
With `entity_spec.get('optional_scope', [])` you didn't need the surrounding if.

Reply:
I was copying the code above, refactored both.